app/renderer/components/download/downloadsBar.js

```diff
@@ -44,36 +50,40 @@ class DownloadsBar extends React.Component {
   mergeProps (state, ownProps) {
     const props = {}
     // used in renderer
+    props.showToolbarWhenDownloading = getSetting(settings.SHOW_TOOLBAR_WHEN_DOWNLOADING)
     props.downloads = downloadUtil.getDownloadItems(state) || Immutable.List()
 
     return props
   }
 
   render () {
-    return <div className='downloadsBar'
+    return <div className={cx({
+      [css(styles.downloadsBar, !this.props.showToolbarWhenDownloading && styles.downloadBar_hidden)]: true,
```

Comment:
Once possible enhancement- if this setting is true (ex: hide toolbar), it would be better to not show the toolbar in the first place.  To do this, you can remove the check here and instead, ONLY set the value in windowReducer.js/menu.js if that value is true. That prevents the component from being created in the first place

Reply:
That sounds rational. As I am not sure how to implement that, would you mind adding a commit for that? thanks 😄